static/app/utils/events.tsx

```diff
@@ -228,12 +229,51 @@ function hasTrace(event: Event) {
  * by ensuring that every inApp frame has a valid sourcemap
  */
 export function eventHasSourceMaps(event: Event) {
-  const inAppFrames = getInAppFrames(event);
+  const inAppFrames = getFrames(event, true);
 
   // the map field tells us if it's sourcemapped
   return inAppFrames.every(frame => !!frame.map);
 }
 
+/**
+ * Function to determine if an event has been symbolicated. If the event
+ * goes through symbolicator, it looks for at least one in-app frame to be successfully
+ * symbolicated. Otherwise falls back to checking for `rawStacktrace` field presence.
+ */
+export function eventIsSymbolicated(event: Event) {
+  const frames = getFrames(event, false);
+  const fromSymbolicator = frames.some(frame => defined(frame.symbolicatorStatus));
+
+  if (fromSymbolicator) {
+    // if the event goes through symbolicator, we say it's symbolicated if at least one in-app
+    // frame is successfully symbolicated
+    return frames.some(
+      frame => frame.inApp && frame.symbolicatorStatus === SymbolicatorStatus.SYMBOLICATED
```

Comment:
Some events don't have `inApp` at all (an OS crash, or caused by 3rd parties), I wonder if in this case, we should still check if there's at least 1 `SYMBOLICATED`.

Right now an event without `inApp` will be considered not symbolicated but might have some system/3rd party frames symbolicated.

Reply:
yup, makes sense, https://github.com/getsentry/sentry/pull/50139/commits/0f7fd1f165b2484c1b77d7df8a2a4ab7fe31ab17